doc/features.txt

```diff
@@ -14,3 +14,16 @@ Currently available feature flags are:
 
     This only impacts the UI, the <source> answerfile construct still
     allows to include supplemental packs without this feature flag.
+
+  large-block-capable-sr-type
+
+    Allow the use of an SR type for local storage that (unlike "lvm" and
+    "ext") is not restricted to using disks with 512 byte blocks. The flag
+    file's content must be the name of the SR type.
+
+    The other expectations for such an SR type are:
+    - Its only non-optional creation parameter (as with "lvm" and "ext") is
+      "device", a comma-separated list devices to use.
+    - It should use an identifying prefix for the names of volume groups that
+      it creates, and this prefix should start "XSLocal". (e.g. if the SR type
+      were called "foo", it might use the prefix "XSLocalFOO".)
```

Comment:
This hardly qualifies as a "feature flag", limits the capability to support large blocks only to a single SR type other, and forbids EXT and LVM to support it in the future.
What about rather a config file containing list of filesystems?

Reply:
If both ext and lvm supported large blocks, then this "flag" could be left unset, and nothing should block their use.

That said, I don't think anything I've done has tied our hands in such a way as would prevent us changing our minds about what to do at a later date.

If the options of what SR types were available and when they could be used were more complicated than lvm & ext + one optional large block SR type then I think there'd need to be a rethink about how the UI for choosing SR type would work.